server/src/main/java/io/druid/server/router/TieredBrokerHostSelector.java

```diff
@@ -312,11 +313,21 @@ Server pick()
         return null;
       }
 
-      if (roundRobinIndex >= currNodes.size()) {
-        roundRobinIndex %= currNodes.size();
+      int index = roundRobinIndex.get();
+      int nextIndex = index + 1;
+
+      while (true) {
```

Comment:
Could you please extract this block as a method, it will allow to not repeat
```java
int index = roundRobinIndex.get();
int nextIndex = index + 1;
```

block

Reply:
sure